codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/Waiters.java

```diff
@@ -0,0 +1,835 @@
+        String waiterClientName = generateWaiterClientName(waiterName);
+
+        StructureShape inputShape = model.expectShape(
+                operationShape.getInput().get(), StructureShape.class
+        );
+        StructureShape outputShape = model.expectShape(
+                operationShape.getOutput().get(), StructureShape.class
+        );
+
+        Symbol inputSymbol = symbolProvider.toSymbol(inputShape);
+        Symbol outputSymbol = symbolProvider.toSymbol(outputShape);
+
+        writer.write("");
+        writer.writeDocs(
+                String.format("%s are waiter options for %s", optionsName, waiterClientName)
+        );
+
+        writer.openBlock("type $L struct {", "}",
+                optionsName, () -> {
+                    writer.addUseImports(SmithyGoDependency.TIME);
+
+                    writer.write("");
+                    writer.writeDocs("MinDelay is the minimum amount of time to delay between retries in seconds");
+                    writer.write("MinDelay time.Duration");
+
+                    writer.write("");
+                    writer.writeDocs("MaxDelay is the maximum amount of time to delay between retries in seconds");
+                    writer.write("MaxDelay time.Duration");
+
+                    writer.write("");
+                    writer.writeDocs("MaxWaitTime is the maximum amount of wait time in seconds before a waiter is "
+                            + "forced to return");
+                    writer.write("MaxWaitTime time.Duration");
+
+                    writer.write("");
+                    writer.writeDocs("MaxAttempts is the maximum number of attempts to fetch a terminal waiter state");
+                    writer.write("MaxAttempts int64");
+
+                    writer.write("");
+                    writer.writeDocs("EnableLogger is used to enable logging for waiter retry attempts");
+                    writer.write("EnableLogger bool");
+
+                    writer.write("");
+                    writer.writeDocs(
+                            "WaiterStateMutator is mutator function option that can be used to override the "
+                                    + "service defined waiter-behavior based on operation output, or returned error. "
+                                    + "The mutator function is used by the waiter to decide if a state is retryable "
+                                    + "or a terminal state.\n\nBy default service-modeled waiter state mutators "
+                                    + "will populate this option. This option can thus be used to define a custom "
+                                    + "waiter state with fall-back to service-modeled waiter state mutators."
+                    );
+                    writer.write(
+                            "WaiterStateMutator func(context.Context, $P, $P, error) "
+                                    + "(bool, error)", inputSymbol, outputSymbol);
```

Comment:
This isn't really a a waiter state mutator, since it doesn't change the states. This option checks the response to determine behavior of the waiter.

The intention of this option is a function that determines if the response puts the waiter into a fail state (exit with error), success state (exit without error), or retry state (can retry).

I think a name like `ContinueWaiting`, or `Retryable` would be more clear of the intention and capabilities. Since the `bool` returned specifies if the waiter could continue to retry, and the `error` is for some failure state. I think `Retryable` would make this more similar to the design of the SDK's attempt retryer.